test/edu/stanford/dlss/wowza/TestSulWowza.java

```diff
@@ -733,6 +735,45 @@ public void verifyTokenAgainstStacksService_validUrl()
     }
 
     @Test
+    public void verifyTokenAgainstStacksService_filenameNeedsEncoding()
+    {
+        WMSProperties mockProperties = mock(WMSProperties.class);
+        when(mockProperties.getPropertyStr(anyString(), anyString())).thenReturn(SulWowza.DEFAULT_STACKS_TOKEN_VERIFICATION_BASEURL);
+        IApplicationInstance appInstanceMock = mock(IApplicationInstance.class);
+        when(appInstanceMock.getProperties()).thenReturn(mockProperties);
+        testModule.onAppStart(appInstanceMock);
+        String druid = "oo000oo0000";
+        String filename = "file name with spaces; and an unusual char @ then end.ext!";
+        String userIp = "0.0.0.0";
+        String expPath = "/media/" + druid + "/" + SulWowza.urlEncode(filename) + "/verify_token";
+        String expQueryStr = "?stacks_token=" + stacksToken + "&user_ip=" + userIp;
+        String expUrlStr = SulWowza.DEFAULT_STACKS_TOKEN_VERIFICATION_BASEURL + expPath + expQueryStr;
+
+        URL resultUrl = testModule.getVerifyStacksTokenUrl(stacksToken, druid, filename, userIp);
+        assertEquals(expUrlStr, resultUrl.toString());
```

Comment:
i kind of like that it also implies that the non-special (regular?) characters are left un-encoded, since the druid and IP are also included in the check.  i could add a comment to that effect.  i could also change it to a substring approach if you'd strongly prefer, though the current approach is my preference at the moment.